### PR TITLE
feat(aws): Add new CodeBuild check to validate environment variables

### PR DESCRIPTION
**Pull request by danibarranqueroo**
### Context

To enhance security and follow best practices for handling sensitive information, we are adding a new check within the CodeBuild framework. This check ensures that sensitive environment variables (AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY) in CodeBuild projects are not stored as clear text credentials.

To facilitate this check, modifications were necessary in the CodeBuild service because the `batch_get_projects` method previously did not utilize this parameter. Additionally, I have renamed all the methods according to the Python naming convention.

In addition, I did not use Moto for testing because the way it creates a new CodeBuild project seems not to accept environment variables, or something similar, because when I debug, they are always empty. I have tested the check using the AWS CLI and it seems to be working great, so for the test file, I had use MagicMock instead of Moto.

### Description

I have added this new check called `codebuild_project_no_plaintext_credentials` with his respective test to prove that it works.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/services/codebuild/codebuild_project_no_plaintext_credentials/codebuild_project_no_plaintext_credentials.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    def execute(self):
+        findings = []
+        for project in codebuild_client.projects.values():
+            report = Check_Report_AWS(self.metadata())
+            report.region = project.region
+            report.resource_id = project.name
+            report.resource_arn = project.arn
+            report.status = "PASS"
+            report.status_extended = f"CodeBuild project {project.name} does not have sensitive environment plaintext credentials."
+            sensitive_vars = ["AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY"]
```
Comment (jfagoagas) [inline prowler/providers/aws/services/codebuild/codebuild_project_no_plaintext_credentials/codebuild_project_no_plaintext_credentials.py]: What do you think about using the `audit_config` to allow users to add/remove sensitive environment variables? It'd be a great addition!

https://docs.prowler.com/projects/prowler-open-source/en/latest/developer-guide/checks/#using-the-audit-config

Comment (danibarranqueroo) [reply prowler/providers/aws/services/codebuild/codebuild_project_no_plaintext_credentials/codebuild_project_no_plaintext_credentials.py]: Good point! I've just added it.

**Comment by MrCloudSec** [comment]
Please use the detect secrets library for this check like in `awslambda_function_no_secrets_in_variables`.

**Comment by jfagoagas** [comment]
> Please use the detect secrets library for this check like in `awslambda_function_no_secrets_in_variables`.

But what is the point of the check? I think is not to have these sensitive environment variables, as per https://docs.aws.amazon.com/securityhub/latest/userguide/codebuild-controls.html#codebuild-2

**Comment by MrCloudSec** [comment]
> > Please use the detect secrets library for this check like in `awslambda_function_no_secrets_in_variables`.
> 
> But what is the point of the check? I think is not to have these sensitive environment variables, as per https://docs.aws.amazon.com/securityhub/latest/userguide/codebuild-controls.html#codebuild-2

We don't have to limit Prowler checks to CIS or SecurityHub controls, if we detect all type of secrets, the check would be more advance and we are also covering the SecurityHub check.

It makes no sense to make the user setting in the `config.yaml` the env variables that he wants to check...

File: prowler/providers/aws/services/codebuild/codebuild_project_no_plaintext_credentials/codebuild_project_no_plaintext_credentials.metadata.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{
+  "Provider": "aws",
+  "CheckID": "codebuild_project_no_plaintext_credentials",
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/codebuild/codebuild_project_no_plaintext_credentials/codebuild_project_no_plaintext_credentials.metadata.json]:
```suggestion
  "CheckID": "codebuild_project_no_secrets_in_variables",
```

**Comment by jfagoagas** [comment]
> the user setting in the `config.yaml` the env variables that he wants to check

@danibarranqueroo I apologise because I explained my point just the opposite as I wanted. What I wanted to say is that it could be helpful for our users to be able to exclude some environment variables via `audit_config` since the Mutelist does not work for this kind of checks since the resource is not the environment variable analysed, you can read more about this here https://github.com/prowler-cloud/prowler/issues/3905

**Comment by danibarranqueroo** [comment]
Finally, after following all your advises, I've completed this check using Detect Secrets to find credentials/secrets in the value of the environmental variables, but only when they are stored as plaintext. Additionally, I've used Audit Config to allow users to exclude specific environment variables from the check by adding their names to the config.yaml.

Take a look at it when you can! 😄 :rocket:

File: prowler/providers/aws/services/codebuild/codebuild_project_no_secrets_in_variables/codebuild_project_no_secrets_in_variables.metadata.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  "Severity": "critical",
+  "ResourceType": "AwsCodeBuildProject",
+  "Description": "This check ensures that AWS CodeBuild projects do not use plaintext AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY as environment variables.",
+  "Risk": "Using plaintext AWS credentials in CodeBuild project environment variables can expose these sensitive keys, leading to unauthorized access and potential security breaches.",
+  "RelatedUrl": "https://docs.aws.amazon.com/codebuild/latest/userguide/how-to-create-project-console.html",
+  "Remediation": {
+    "Code": {
+      "CLI": "",
+      "NativeIaC": "",
+      "Other": "https://docs.aws.amazon.com/securityhub/latest/userguide/codebuild-controls.html",
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/codebuild/codebuild_project_no_secrets_in_variables/codebuild_project_no_secrets_in_variables.metadata.json]:
```suggestion
      "Other": "https://docs.aws.amazon.com/securityhub/latest/userguide/codebuild-controls.html#codebuild-2",
```

File: prowler/providers/aws/services/codebuild/codebuild_project_no_secrets_in_variables/codebuild_project_no_secrets_in_variables.metadata.json [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  "CheckType": [
+    "Security Best Practices"
+  ],
+  "ServiceName": "codebuild",
+  "SubServiceName": "",
+  "ResourceIdTemplate": "arn:partition:service:region:account-id:resource-id",
+  "Severity": "critical",
+  "ResourceType": "AwsCodeBuildProject",
+  "Description": "This check ensures that AWS CodeBuild projects do not use plaintext AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY as environment variables.",
+  "Risk": "Using plaintext AWS credentials in CodeBuild project environment variables can expose these sensitive keys, leading to unauthorized access and potential security breaches.",
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/codebuild/codebuild_project_no_secrets_in_variables/codebuild_project_no_secrets_in_variables.metadata.json]: Do this description generic, this check is not only flagging AWS credentials.

File: prowler/providers/aws/services/codebuild/codebuild_project_no_secrets_in_variables/codebuild_project_no_secrets_in_variables.metadata.json [hunk 5]
```diff
@@ -0,0 +1,32 @@
+{
+  "Provider": "aws",
+  "CheckID": "codebuild_project_no_secrets_in_variables",
+  "CheckTitle": "Ensure CodeBuild projects do not contain plaintext AWS credentials",
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/codebuild/codebuild_project_no_secrets_in_variables/codebuild_project_no_secrets_in_variables.metadata.json]: Do this title generic, this check is not only flagging AWS credentials.

**Review by MrCloudSec — CHANGES_REQUESTED** [review]
Great work @danibarranqueroo ! Please, update the metadata of the check according to the new changes.

**Review by MrCloudSec — APPROVED** [review]
👏🏼

**Pull request merged**